src/web/src/components/Posts/GitHubInfo.tsx

```diff
@@ -45,31 +44,38 @@ const filterGitHubUrls = (urls: string[]) => {
     const matches = /^\/(?<user>[^\/]+)\/(?<repo>[^\/]+)((\/(.*))?\/(?<type>[^\/]+)\/(?<id>(\w+))$)?/i.exec(
       pathname
     );
-    if (matches?.groups === undefined) continue;
+    if (matches?.groups === undefined) {
+      continue;
+    }
     const { type, user, repo } = matches.groups;
 
     const repoUrl = `https://github.com/${user}/${repo}`;
-    !repos.includes(repoUrl) && repos.push(repoUrl);
+    repos.add(repoUrl);
     switch (type?.toLowerCase()) {
       case 'pull':
-        !pullRequests.includes(href) && pullRequests.push(href);
+        pullRequests.add(href);
```

Comment:
Question: why `href` vs. `pathname` everywhere?  Don't you want to strip out query params, etc?  Also, why bother to still include the GitHub `origin` in what you pass down?

Reply:
Co-ordinate with @menghif, but I would.  It's all happening as part of his upstream PR.  Might as well do it now.